### DISCUSSION TITLE
How to manually sync a subscription?

### DISCUSSION THREAD

**Discussion by rathboma** [pay-rails/pay#484] [General] [answered]
Lets say I updated a subscription in the stripe dashboard, but didn't have the webhooks running, so I want to force it to pull changes from stripe, how would I do that?

**Comment by excid3** [comment]
You can use `sync`
https://github.com/pay-rails/pay/blob/a6ef45d758f4c6a6b2803fa40b396d0c11aab2ac/lib/pay/stripe/subscription.rb#L22

```ruby
Pay::Stripe::Subscription.sync("ID")
```

**Reply by mpvosseller** [reply]
@excid3 What if the user created at subscription at Stripe (via Stripe Checkout) and our webhooks were not running / didn't work etc... In this case we don't have a subscription id that we can sync. Is there a way to "sync the user" so as to get their subscription down from Stripe?

**Reply by excid3** [reply]
You can use `sync_subscriptions`

```ruby
@user.payment_processor.sync_subscriptions
```

https://github.com/pay-rails/pay/blob/master/lib/pay/stripe/billable.rb#L194

I believe this only works for Stripe right now.

**Comment by jeanschdev** [comment]
Its possible sync the payment method too?

**Reply by excid3** [reply, marked as answer]
`Pay::Stripe::PaymentMethod.sync(stripe_payment_method_id)`

**Reply by jeanschdev** [reply]
Thanks Chris, but I'm referring to customers who already have their payment method in Stripe and were not created in Jumpstart for some reason. How could I synchronize them?

**Reply by jeanschdev** [reply]
Sorry i understand now, the id is the payment method id on stripe, thanks!

**Reply by excid3** [reply]
Yep! Updated my example that more obvious. 👍
